File: src/main/java/org/cerion/stocklist/repository/DividendRepository.java
package org.cerion.stocklist.repository;

import org.cerion.stocklist.model.Dividend;
import org.cerion.stocklist.model.HistoricalDates;

import java.util.List;

public interface DividendRepository {
    List<Dividend> get(String symbol);
    HistoricalDates getHistoricalDates(String symbol);
    void add(String symbol, List<Dividend> list);
    void deleteAll();
}

File: src/main/kotlin/repository/DividendRepository.kt
package org.cerion.stocks.core.repository

import org.cerion.stocks.core.model.Dividend
import org.cerion.stocks.core.model.HistoricalDates

interface DividendRepository {
    fun get(symbol: String): List<Dividend>
    fun getHistoricalDates(symbol: String): HistoricalDates?
    fun add(symbol: String, list: List<Dividend>)
    fun deleteAll()
}
